### ISSUE TITLE
`already_authenticated` should set flash `notice` instead of flash `alert`

### ISSUE THREAD

**Issue by crispinheneise** [heartcombo/devise#5302]
https://github.com/heartcombo/devise/blob/83a32e6d2118b0535cf54b48df4f9853d85b55fd/app/controllers/devise_controller.rb#L114-L117

This sets the `flash[:alert]`, which in most apps is an error message. This is also evidenced by the `already_authenticated` key being set in the `devise.failure` scope. 

In most apps, failure or alert messages are displayed with red text indicating that something went wrong. In my opinion this feels strange, because if you're already signed in, nothing went wrong. In fact, nothing is wrong, you just tried to sign in again unnecessarily. 

It seems like it would be trivial to move the `already_authenticated` message from `devise.failure` to `devise.sessions` and set the flash key to `notice` instead of `alert`. 

Would you be open to a change like this? If so, I can submit a PR.

**Comment by johansenja** [comment]
I think it would be neat to have an option to skip showing the message entirely, because for me this doesn't add that much to the UX (most users probably wouldn't care if they're already signed in, but just want to use the app). Equally open to submitting a quick PR to enable this

**Comment by timhaines** [comment]
@johansenja that makes a lot of sense to me.  This alert is very odd.

**Comment by crispinheneise** [comment]
I have introduced PR #5398 to change the flash message from 'alert' to 'notice'. Do you think this is the best way to do this? Should the flash level be configurable instead of hard-coded? Your feedback is requested.

**Comment by crispinheneise** [comment]
> I think it would be neat to have an option to skip showing the message entirely, because for me this doesn't add that much to the UX (most users probably wouldn't care if they're already signed in, but just want to use the app). Equally open to submitting a quick PR to enable this

@johansenja I agree it doesn't add much to the UI. Failing silently would be a good option. It seems like adding an option for this would be pretty heavy, since it would require: 
- adding a `mattr_accessor` in `lib/devise.rb`
- adding documentation in the README
- adding a line in the template for `config/initializers/devise.rb`
- adding tests

Maybe there is an easier way? Looking at the `set_flash_message` method, it looks like it should be possible to simply blank out the `already_authenticated` key in `devise.en.yml`, since it seems to only set the flash if the message is present: 

https://github.com/heartcombo/devise/blob/c82e4cf47b02002b2fd7ca31d441cf1043fc634c/app/controllers/devise_controller.rb#L154-L161

However in my very quick experiment doing this resulted in a "translation missing" error. 

Do you think it's still worth adding a config option for this?
